### ISSUE TITLE
Displays no watt number

### ISSUE THREAD

**Issue by Xfox0ne** [0xMartin/energy.monitor#3]
It displays and visualizes the battery percentage, but not the total power consumption. Another widget monitors the CPU power usage, but it requires you to authorize it to do so. Maybe that is the problem?

OS: Manjaro Linux x86_64
Host: 81SX Legion Y540-15IRH
Kernel: 6.0.0-1-MANJARO
DE: Plasma 5.25.5
WM: KWin

**Comment by ZGLinus** [comment]
Maybe it isn't like that, the original code is using the voltage and current to get the power. Some systems provide the power instead of current so that the power couldn't display correctly.
The temporary solution is the using the code showed below to replace the file `/home/<User name>/.local/share/plasma/plasmoids/energy.monitor/contents/ui/power.js` and reboot your system.
```
/*
 * Copyright 2021 Martin Krcma <martin.krcma1@gmail.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of
 * the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http: //www.gnu.org/licenses/>.
 * 
 */

function getBatteryPath() {
    for (var i = 0; i < 4; i++) {
        var req = new XMLHttpRequest();
        req.open("GET", "/sys/class/power_supply/BAT" + i + "/voltage_now", false);
        req.send(null);
        if (req.responseText != "") return "/sys/class/power_supply/BAT" + i;
    }
    return "";
}

function getPower(fileUrl) {
    if (fileUrl == "") return 0.0;

    var curReq = new XMLHttpRequest();
    curReq.open("GET", fileUrl + "/power_now", false);
    curReq.send(null);

    var power = (parseInt(curReq.responseText)) / 1000000;
    if (Number.isNaN(power)) return 0.0;
    return Math.round(power * 10) / 10;

}
```
Forgive my grammar and spelling mistakes because English isn't my mother tongue.

**Comment by 0xMartin** [comment]
This bug will be fixed in the next update 1.2 which will be released soon.
